File: ckgame-java/src/main/java/com/ckgame/events/EventChoice.java
package com.ckgame.events;

import java.util.List;

/**
 * 事件选项。
 */
public final class EventChoice {
    public final int id;
    public final String text;
    public final List<Effect> effects;
    public final double aiWeight;

    public EventChoice(int id, String text, List<Effect> effects, double aiWeight) {
        this.id = id;
        this.text = text;
        this.effects = effects;
        this.aiWeight = aiWeight;
    }

    public EventChoice(int id, String text, List<Effect> effects) {
        this(id, text, effects, 5.0);
    }
}

File: ckgame-ts/src/events/EventChoice.ts
import { Effect } from './Effect.js';

/**
 * An event choice.
 */
export class EventChoice {
  public readonly id: number;
  public readonly text: string;
  public readonly effects: Effect[];
  public readonly aiWeight: number;

  constructor(id: number, text: string, effects: Effect[], aiWeight: number = 5) {
    this.id = id;
    this.text = text;
    this.effects = effects;
    this.aiWeight = aiWeight;
  }
}
